src/mito2/src/manifest/manager.rs

```diff
@@ -373,20 +306,54 @@ impl RegionManifestManagerInner {
         Ok(version)
     }
 
+    /// Retrieves the current [RegionManifest].
+    pub fn manifest(&self) -> Arc<RegionManifest> {
+        self.manifest.clone()
+    }
+
     /// Returns total manifest size.
-    pub(crate) fn total_manifest_size(&self) -> u64 {
+    pub fn manifest_usage(&self) -> u64 {
         self.store.total_manifest_size()
     }
-}
 
-impl RegionManifestManagerInner {
+    /// Returns true if a newer version manifest file is found.
+    ///
+    /// It is typically used in read-only regions to catch up with manifest.
+    pub async fn has_update(&self) -> Result<bool> {
+        let last_version = self.last_version;
+
+        let streamer =
+            self.store
+                .manifest_lister()
+                .await?
+                .context(error::EmptyManifestDirSnafu {
+                    manifest_dir: self.store.manifest_dir(),
+                })?;
+
+        let need_update = streamer
+            .try_any(|entry| async move {
+                let file_name = entry.name();
+                if is_delta_file(file_name) {
+                    let version = file_version(file_name);
+                    if version > last_version {
+                        return true;
+                    }
+                }
+                false
+            })
+            .await
+            .context(error::OpenDalSnafu)?;
+
+        Ok(need_update)
+    }
+
     /// Increases last version and returns the increased version.
     fn increase_version(&mut self) -> ManifestVersion {
         self.last_version += 1;
         self.last_version
     }
 
-    pub(crate) async fn may_do_checkpoint(&mut self, version: ManifestVersion) -> Result<()> {
+    async fn may_do_checkpoint(&mut self, version: ManifestVersion) -> Result<()> {
```

Comment:
Adds some doc for this function, because it's `public` right now.

Reply:
> because it's `public` right now.

It was public before. This PR changed it to private.